pytensor/tensor/rewriting/subtensor.py

```diff
@@ -1354,6 +1376,7 @@ def local_useless_inc_subtensor_alloc(fgraph, node):
                     z_broad[k]
                     and not same_shape(xi, y, dim_x=k, dim_y=k)
                     and shape_of[y][k] != 1
+                    and shape_of[xi][k] == 1
```

Comment:
If this fixes a bug, it will need a specific regression test, and be in a separate commit